packages/git-diff/lib/git-diff-view.js

```diff
@@ -1,64 +1,143 @@
-const { CompositeDisposable } = require('atom');
-const { repositoryForPath } = require('./helpers');
+'use babel';
+
+import { CompositeDisposable } from 'atom';
+import repositoryForPath from './helpers';
 
 const MAX_BUFFER_LENGTH_TO_DIFF = 2 * 1024 * 1024;
 
-module.exports = class GitDiffView {
+/**
+ * @describe Handles per-editor event and repository subscriptions.
+ * @param editor {Atom.TextEditor} - The editor this view will manage.
+ */
+export default class GitDiffView {
   constructor(editor) {
-    this.updateDiffs = this.updateDiffs.bind(this);
-    this.editor = editor;
+    // These are the only members guaranteed to exist.
     this.subscriptions = new CompositeDisposable();
-    this.decorations = {};
-    this.markers = [];
-  }
+    this.editor = editor;
+    this.repository = null;
+    this.markers = new WeakMap();
```

Comment:
You can use a `Map` and use `clear()` function when they are destoryed.

Reply:
But why add in extra effort if the garbage collector will already be running?

Reply:
> But why add in extra effort if the garbage collector will already be running?

For style. IMO it'd add clarity, telling one reading the code that you are not going to use the Map anymore

Reply:
True. I prefer this way because it doesn't block the render pipeline. I want the draw function to run as fast as possible. I was thinking about making a future contribution which optimized redraws by modifying the cached set of diffs instead of regenerating it every redraw which would cut down on memory releases within the `WeakMap`; the diffs can't be helped because that's an API issue with `git-utils`, but the markers can. By simply swapping the changed diffs within it's array, the `WeakMap` removes the corresponding markers without extra memory management. This method makes it easier for me to write those future modifications.

Reply:
`WeakMap` does not guarantee the removal! We have a hard requirement for removing the entries when they are destroyed, otherwise, we will keep the markers that are destroyed!

See this video to understand more about weak references:
https://youtu.be/uygxJ8Wxotc

Reply:
Thanks for the informative videos. Really appreciated! I'll go ahead and change that to a `Map` then.